File: backend/src/main/java/com/corpomotriz/repuestos/model/Proveedor.java
package com.corpomotriz.repuestos.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "proveedor", schema = "repuestos_db")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
@ToString
public class Proveedor {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombre;
    private String telefono;
    private String email;
    private String localizacion;
    private String especializacion;
    private String plazoEntrega;

    @Column(name = "fecha_creacion")
    private LocalDateTime fechaCreacion;

    @OneToMany(mappedBy = "proveedor")
    private List<Producto> productos;

}

File: frontend/src/app/proveedor.ts
export class Proveedor {
  id: number;
  nombre: string;
  telefono: string;
  email: string;
  localizacion: string;
  especializacion: string;
  plazoEntrega: string;
  fechaCreacion: string;
}

export interface ProveedorCrearDTO{
  nombre: string;
  telefono: string;
  email: string;
  localizacion: string;
  especializacion: string;
  plazoEntrega: string;
  fechaCreacion: string;
  id: number;
}